grommet/grommet
# Changed to allow hoverindicator to accept background props

#### What does this PR do?
This PR allows for HoverIndicator prop to accept the same types that background accepts for box since they are both working the same under the hood
#### Where should the reviewer start?
Button/doc.js
#### What testing has been done on this PR?
Branched out grommet with a local project to test the changes being made with checking both prop types as well as TS errors 
#### How should this be manually tested?
```

 <Box width="medium" align="center" gap="large">
      <Text>Button with object passed to `hoverIndicator` prop:</Text>
      <Button
        hoverIndicator={{
          // color: "brand",
          // opacity: "weak",
          // dark: "true"
         // {{"dark": "light-2", "light": "dark-2"}}
          color: "brand",
          image:
            "url(http://librelogo.org/wp-content/uploads/2014/04/gradient2.png)"
        }}
        label="Submit"
        onClick={() => {}}
      />
      <Text>Box with same object passed to `background` prop:</Text>
      <Box
        height="small"
        width="small"
        align="center"
        background={{
          color: "brand",
          image:
            "url(http://librelogo.org/wp-content/uploads/2014/04/gradient2.png)"
        }}
        // color: "brand",
        // opacity: "weak",
        // dark: "true"
       // {{"dark": "light-2", "light": "dark-2"}}
      />
    </Box>

```

First by testing that all of the props that work for background box also can be used for hoverindicator and work as well. Then you can test the same code example in TS to make sure that the different props are also being accepted 

#### Any background context you want to provide?

#### What are the relevant issues?
closes issue #3287 
#### Screenshots (if appropriate)

#### Do the grommet docs need to be updated?
no
#### Should this PR be mentioned in the release notes?
no
#### Is this change backwards compatible or is it a breaking change?
backwards compatible
